### DISCUSSION TITLE
How to add additional include paths in compile_commands.json or equivalent?

### DISCUSSION THREAD

**Discussion by erusyd** [emacs-lsp/lsp-mode#3300] [Q&A] [answered]
I'm developing embedded c/c++ applications using eclipse which uses CMake to build projects. I can set CMake to generate compile_commands.json, but it seems that it does not include all the necessary include paths and thus emacs lsp show "xxx/xxx.h" file not found.

So my question is:
How to add those missing paths into compile_commands.json to assist lsp-mode to index header files? Or is there any other mechanism to support lsp-mode to find extra include path?

**Comment by yyoncho** [comment, marked as answer]
clangd/ccls does not expose this via API. The only way to make that work that I am aware of is to use compile_commands.json. I guess you can do that manually.

**Reply by erusyd** [reply]
Thanks for your reply.  Give up to manually add those paths.
